File: src/domain/storage-stack.js
export function moveItemToTop(stack, itemId) {
  return moveItemToPosition(stack, itemId, 1);
}

export function moveItemToPosition(stack, itemId, oneBasedPosition) {
  const currentIndex = stack.indexOf(itemId);
  if (currentIndex === -1) {
    return [...stack];
  }

  const nextStack = stack.filter((id) => id !== itemId);
  const targetIndex = Math.max(0, Math.min(nextStack.length, oneBasedPosition - 1));
  nextStack.splice(targetIndex, 0, itemId);
  return nextStack;
}

File: lib/domain/storage_stack.dart
List<String> moveItemToTop(List<String> stack, String itemId) {
  return moveItemToPosition(stack, itemId, 1);
}

List<String> moveItemToPosition(List<String> stack, String itemId, int oneBasedPosition) {
  final currentIndex = stack.indexOf(itemId);
  if (currentIndex == -1) {
    return [...stack];
  }

  final nextStack = stack.where((id) => id != itemId).toList();
  final targetIndex = (oneBasedPosition - 1).clamp(0, nextStack.length);
  nextStack.insert(targetIndex, itemId);
  return nextStack;
}